compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -18,13 +18,10 @@ fn resolve_instance<'tcx>(
 
     let result = if let Some(trait_def_id) = tcx.trait_of_item(def) {
         debug!(" => associated item, attempting to find impl in param_env {:#?}", param_env);
-        resolve_associated_item(
-            tcx,
-            def,
-            param_env,
-            trait_def_id,
-            tcx.normalize_erasing_regions(param_env, substs),
-        )
+        match tcx.try_normalize_erasing_regions(param_env, substs) {
+            Ok(substs) => resolve_associated_item(tcx, def, param_env, trait_def_id, substs),
+            Err(_error) => return Err(ErrorGuaranteed::unchecked_claim_error_was_emitted()),
```

Comment:
you should really never be using `ErrorGuaranteed::unchecked_claim_error_was_emitted()`, it's pretty easy to introduce bugs that could affect the compiler's soundness --

From the docs:

> To be used only if you really know what you are doing…